src/collector/collector.rs

```diff
@@ -23,7 +27,7 @@ pub(super) trait Collector {
     /// as part of the collector registration and only then feed the collector
     /// with data coming from the core. Checks for the mandatory part of the
     /// collector should be done here.
-    fn init(&mut self, kernel: &mut probe::Kernel) -> Result<()>;
+    fn init(&mut self, kernel: &mut probe::Kernel, cli: &CliConfig) -> Result<()>;
```

Comment:
I was thinking maybe we don't need a macro. We cannot add templated functions on SubCommand Trait (because it makes it non object-safe) but we can add them to CliConfig, e.g:

impl CliConfig {
    pub(crate) fn get_section<T>(&self, name: &str) -> Result<M> {
         self.subcommand.dynamic().get_section()
    }
}

Reply:
Done,